R/GeospatialFunctions.R

```diff
@@ -0,0 +1,571 @@
+#' TADA_MakeSpatial
+#' 
+#' Transform a Water Quality Portal dataframe into a geospatial {sf} object.
+#' 
+#' @param data A dataframe created by `TADA_DataRetrieval()`.
+#' @param crs The coordinate reference system (CRS) you would like the returned point features to be in. The default is CRS 4326 (WGS84).
+#' 
+#' @return The original TADA Water Quality Portal dataframe but as geospatial {sf} point objects. 
+#' 
+#' @seealso [TADA_DataRetrieval()]
+#' 
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' tada_not_spatial <- TADA_DataRetrieval(startDate = "1995-01-01",
+#'                                        endDate = "1995-12-31",
+#'                                        characteristicName = "pH",
+#'                                        statecode = "SC", 
+#'                                        countycode = "Abbeville",
+#'                                        applyautoclean = TRUE)
+#' 
+#' # make `tada_not_spatial` an {sf} object, projected in crs = 4269 (NAD83)                                 
+#' tada_spatial <- TADA_MakeSpatial(data = tada_not_spatial, crs = 4269)
+#' }
+
+TADA_MakeSpatial <- function(data, crs = 4326){
+  
+  # must have all columns needed to make spatial
+  if (!"LongitudeMeasure" %in% colnames(data) & !"LatitudeMeasure" %in% colnames(data)) {
```

Comment:
Although unlikely, if only one of the columns is missing for some reason this step won't catch it. So you might consider something like an `|` instead of `&` here.

Reply:
tested this and I agree with MB